src/phillips_dessouky.rs

```diff
@@ -0,0 +1,84 @@
+use pyo3::prelude::*;
+
+use ordered_float::OrderedFloat;
+use pathfinding::directed::edmonds_karp::{
+    SparseCapacity,
+    Edge,
+    edmonds_karp
+};
+
+use std::time::{
+    Instant,
+    Duration,
+};
+use log::info;
+
+
+#[pyclass]
+pub struct PhillipsDessouky {
+    #[pyo3(get)]
+    node_ids: Vec<u32>,
+    source_node_id: u32,
+    sink_node_id: u32,
+    edges_raw: Vec<((u32, u32), f64)>,
+}
+
+#[pymethods]
+impl PhillipsDessouky {
+    #[new]
+    fn new(
+        node_ids: Vec<u32>,
+        source_node_id: u32,
+        sink_node_id: u32,
+        edges_raw: Vec<((u32, u32), f64)>,
+    ) -> PyResult<Self> {
+        Ok(PhillipsDessouky {
+            node_ids,
+            source_node_id,
+            sink_node_id,
+            edges_raw,
+        })
+    }
+
+    fn max_flow(&self) -> PyResult<Vec<((u32, u32), f64)>> {
+        let profiling_start = Instant::now();
+        let edges_edmonds_karp: Vec<Edge<u32, OrderedFloat<f64>>> = self.edges_raw.iter().map(|((from, to), cap)| {
+            ((*from, *to), OrderedFloat(*cap))
+        }).collect();
+        let profiling_end = Instant::now();
+        info!("PROFILING Rust_PhillipsDessouky::max_flow scaling to OrderedFloat time: {:.10}s", PhillipsDessouky::profile_duration(profiling_start, profiling_end));
+
+        let profiling_start = Instant::now();
+        let (flows, _max_flow, _min_cut) = edmonds_karp::<_, _, _, SparseCapacity<_>>(
+            &self.node_ids,
+            &self.source_node_id,
+            &self.sink_node_id,
+            edges_edmonds_karp,
+        );
+        let profiling_end = Instant::now();
+        info!("PROFILING Rust_PhillipsDessouky::max_flow edmonds_karp time: {:.10}s", PhillipsDessouky::profile_duration(profiling_start, profiling_end));
+
+        let profiling_start = Instant::now();
+        let flows_f64: Vec<((u32, u32), f64)> = flows.iter().map(|((from, to), flow)| {
+            ((*from, *to), flow.into_inner())
+        }).collect();
```

Comment:
```suggestion
        let flows_f64: Vec<((u32, u32), f64)> = flows.into_iter().map(|((from, to), flow)| {
            ((from, to), flow.into_inner())
        }).collect();
```
I think this copy can be avoided. Well, before you tell me that this doesn't compile ;)

Reply:
Updated a little bit. I don't think `edge` has to be unpacked into `(from, to)` as well.